proposed/ObjectToObjects.md

```diff
@@ -0,0 +1,56 @@
+Object To Objects
+=================
+
+Summary
+-------
+
+Objects are always passed by reference, not by value. This means you always have shared state when passing objects around.
+Shared state is one of the main culprits for unnecessary or incidental complexity. Complexity is bad.
+
+re-But...tals
+-------------
+
+- "But I use value objects!" Good for you, that is not actually OOP though.
+
+- "But all my objects are immutable." Yup, same as value objects. See above.
+
+- "OOP is the future!". Carnegie Melloin disagrees, they don't actually offer OOP as a course anymore. OO Design does 
+  have merit. But you should probably check out this Functional Programming thing.
+
+In the wild
+-----------
+
+- DateTime, part of PHP. There is now also an ImmutableDateTime. Guess why.
```

Comment:
Links should be added to these entries in PHP manual, for the sake of ease-of-use and education.